src/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,177 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { IconDefinition } from "@fortawesome/free-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { ThemeContext, ThemeProvider } from "../../theme";
+import { isNavbarItem } from "./Navbar.utils";
+import { isDev } from "../../utils";
+import useDotAnimation from "./Navbar.hooks";
+import style from "../../variables.module.scss";
+import styles from "./navbar.module.scss";
+import classNames from "classnames/bind";
+
+const cx = classNames.bind(styles);
+
+const NavbarContext = React.createContext<{
+  items?: React.RefObject<HTMLLIElement>[];
+  activeId?: number;
+  setActiveId?: (id?: number) => void;
+  registerItem?: (item: React.RefObject<HTMLLIElement>) => void;
+  unRegisterItem?: (item: React.RefObject<HTMLLIElement>) => void;
+}>({});
+
+type NavbarItemProps<T extends string> = {
+  as: "a" | "button";
+  href?: string;
+  icon: IconDefinition;
+  title: T;
+  onSelect: ({ title, id }: { title: T; id?: number }) => void;
+  id?: number; // TODO: find out how not to expose this prop
+  iconStyles?: React.CSSProperties;
+};
+
+export function NavbarItem<T extends string>({
+  as: Comp,
+  href,
+  icon,
+  title,
+  onSelect,
+  id,
+  iconStyles,
+}: NavbarItemProps<T>) {
+  const theme = useContext(ThemeContext);
+  const { activeId, setActiveId, registerItem, unRegisterItem } =
+    useContext(NavbarContext);
+  const itemRef = useRef<HTMLLIElement>(null);
+
+  const handleClick = () => {
+    if (id !== activeId) {
+      setActiveId?.(id);
+      onSelect({ title, id });
+    }
+  };
+
+  useEffect(() => {
+    registerItem?.(itemRef);
+    return () => unRegisterItem?.(itemRef);
+  }, [setActiveId, id, registerItem, unRegisterItem]);
+
+  const isActive = id === activeId;
+
+  const additionalProps = Comp === "a" ? { href } : {};
+
+  return (
+    <li className={styles.navbarItem} ref={itemRef}>
+      <Comp
+        {...additionalProps}
+        onClick={handleClick}
+        className={styles.navbarItemBtn}
+      >
+        <span className={styles.navbarItemIconContainer}>
+          <FontAwesomeIcon
+            data-test-id={`${icon.iconName}-icon`}
+            className={cx("navbarItemIcon", { active: isActive })}
+            style={iconStyles}
+            icon={icon}
+            color={
+              isActive
+                ? iconStyles?.color ?? style.primary
+                : theme?.colorMode === "dark"
```

Comment:
todo: You don't need JS for this theme conditional. Check the [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) for CSS ;)

Reply:
nice 😎 updated

Reply:
sweeet ✨